pyproject.toml

```diff
@@ -0,0 +1,17 @@
+[project]
+name = "promdens"
+version = "0.1.0"
+description = "Promoted Density Approach for sampling initial conditions for trajectory-based nonadiabatic photodynamics"
+readme = "README.md"
+requires-python = ">=3.7"
+dependencies = [
+    "numpy>=1.15",
+    "matplotlib~=3.0",
+]
+
+[project.scripts]
+promdens = "promdens:main"
+
+[build-system]
+requires = ["hatchling"]
+build-backend = "hatchling.build"
```

Comment:
To clarify this `build-backend` business a bit: 

Python packaging differentiates between build "backends" and "frontends". Build frontends are installers like pip or uv, which have a CLI that you directly interact with as a user. 

build backend is the program / library that does the actual work of packing / extracting files, compiling C extensions if any etc. The most common example of this is `setuptools`. `hatchling` is a more modern version, but they all do basically the same thing.

@JanosJiri not sure if this is helpful or not. If you find all of this confusing, you're very much not alone. It took me a loong time to orient myself in the holy mess of Python packaging. :sweat_smile: 

Here's a nice guide in case you're interested in details https://packaging.python.org/en/latest/tutorials/packaging-projects/